### PR TITLE
Release: 1.2.0

### PR DESCRIPTION
**Pull request by timmyc**
Version 1.2.0 is what will be released as part of WooCommerce 4.2 in June.

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Dev: Add Profiler Step View Tracks #4141
+- Fix: Updated messaging after last step in OBW. #4148
+- Performance: trim down inbox note API request. #3977
+- Tweak: Update Email Marketing note. #4167
+- Fix: Reset profiler when visiting old OBW URL #4166.
+- Tweak: Adjust "demo products" verbiage to "Sample Products" #4184 🎉 jobthomas
+- Dev: Add React Testing Library #4221
+- Enhancement: New Link Component #4219
+- Performance: Use Route based code splitting to reduce bundle size #4094
+- Tweak: Use PAGE_ROOT constant to reduce redundant strings #4238 🎉 codemascot
```
Comment (psealock) [inline readme.txt:94]: I'd be nice to get the `@` before the contributor name so they get a notification.

File: readme.txt [hunk 2]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Tweak: Decouple Plugins DataStore from onboarding feature #4048
+- Tweak: Move API out of Onboarding #4093
+- Dev: Add Profiler Step View Tracks #4141
+- Fix: Updated messaging after last step in OBW. #4148
+- Performance: trim down inbox note API request. #3977
+- Tweak: Update Email Marketing note. #4167
+- Fix: Reset profiler when visiting old OBW URL #4166.
+- Tweak: Adjust "demo products" verbiage to "Sample Products" #4184 🎉 jobthomas
+- Dev: Add React Testing Library #4221
+- Enhancement: New Link Component #4219
```
Comment (psealock) [inline readme.txt:92]: This wasn't a new component.
```suggestion
- Dev: Add List and Link components to Storybook #4219
```

**Review by psealock — APPROVED** [review]
Looking good @timmyc with a few minor comments.

Should we take the opportunity to move to an RC-* release process?

File: readme.txt [hunk 3]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
 
 == Changelog ==
 
+= 1.2.0 2020-05--8 =
+- Fix: Cast Shipping Total to float #4042 🎉barryhughes
```
Comment (mattsherman) [inline readme.txt:75]: This is a Dev change, as it isn't visible to end users.

```suggestion
- Dev: Cast Shipping Total to float #4042 🎉barryhughes
```

File: readme.txt [hunk 4]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
 
 == Changelog ==
 
+= 1.2.0 2020-05--8 =
+- Fix: Cast Shipping Total to float #4042 🎉barryhughes
+- Tweak: Dynamic Currency with Context API #4027
```
Comment (mattsherman) [inline readme.txt:76]: This is a Dev change.

```suggestion
- Dev: Dynamic Currency with Context API #4027

```

File: readme.txt [hunk 5]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
 
 == Changelog ==
 
+= 1.2.0 2020-05--8 =
+- Fix: Cast Shipping Total to float #4042 🎉barryhughes
+- Tweak: Dynamic Currency with Context API #4027
+- Fix: Remove Duplicate array entry #4049 🎉 tivnet
```
Comment (mattsherman) [inline readme.txt:77]: This is a change to the examples in docs, so Dev.

```suggestion
- Dev: Remove Duplicate array entry in documentation example #4049 🎉 tivnet
```

File: readme.txt [hunk 6]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+= 1.2.0 2020-05--8 =
+- Fix: Cast Shipping Total to float #4042 🎉barryhughes
+- Tweak: Dynamic Currency with Context API #4027
+- Fix: Remove Duplicate array entry #4049 🎉 tivnet
+- Fix: Proper display of elements in wc-admin pages when in a RTL environment. #4051
+- Enhancement: Marketing Inbox Note #4030
+- Tweak: Enable the default homepage template to be filtered #4072 🎉 stevegrunwell
+- Tweak: Create admin note if Jetpack or WooCommerce Services plugin doesn't get installed due to an error during OBW #3888
+- Fix: Update UX when knowledge base articles fail to retrieve #4133
+- Tweak: Decouple Plugins DataStore from onboarding feature #4048
```
Comment (mattsherman) [inline readme.txt:83]: This is a Dev change.

```suggestion
- Dev: Decouple Plugins DataStore from onboarding feature #4048
```

File: readme.txt [hunk 7]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Tweak: Move API out of Onboarding #4093
```
Comment (mattsherman) [inline readme.txt:84]: This is a Dev change.

```suggestion
- Dev: Move API out of Onboarding #4093
```

File: readme.txt (hunk 1 above)
Comment (mattsherman) [inline readme.txt:94]: This is a Dev change.

```suggestion
- Dev: Use PAGE_ROOT constant to reduce redundant strings #4238 🎉 codemascot

```

File: readme.txt [hunk 8]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Tweak: Adjust "demo products" verbiage to "Sample Products" #4184 🎉 jobthomas
+- Dev: Add React Testing Library #4221
+- Enhancement: New Link Component #4219
+- Performance: Use Route based code splitting to reduce bundle size #4094
+- Tweak: Use PAGE_ROOT constant to reduce redundant strings #4238 🎉 codemascot
+- Enhancement: Add onboarding payments note #4157 
+- Tweak: Don't reschedule imports on failed imports #4263
+- Tweak: Remove obsolete inbox messages #4182
+- Fix: Dashboard flash before OBW chunk loads #4259
+- Fix: Make query selector for admin alerts more specific #4289 🎉pauloiankoski
```
Comment (mattsherman) [inline readme.txt:99]: This is a Dev change.

```suggestion
- Dev: Make query selector for admin alerts more specific #4289 🎉pauloiankoski 
```

File: readme.txt [hunk 9]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Dev: Add React Testing Library #4221
+- Enhancement: New Link Component #4219
+- Performance: Use Route based code splitting to reduce bundle size #4094
+- Tweak: Use PAGE_ROOT constant to reduce redundant strings #4238 🎉 codemascot
+- Enhancement: Add onboarding payments note #4157 
+- Tweak: Don't reschedule imports on failed imports #4263
+- Tweak: Remove obsolete inbox messages #4182
+- Fix: Dashboard flash before OBW chunk loads #4259
+- Fix: Make query selector for admin alerts more specific #4289 🎉pauloiankoski 
+- Fix: Guard against null themes in OBW #4244
```
Comment (mattsherman) [inline readme.txt:100]: This is a Dev change.

```suggestion
- Dev: Guard against null themes in OBW #4244
```

File: readme.txt [hunk 10]
```diff
@@ -71,6 +71,36 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
...
+- Fix: Update wcadmin db version after db callback #4323
```
Comment (mattsherman) [inline readme.txt:101]: This is a Dev change.

```suggestion
- Dev: Update wcadmin db version after db callback #4323
```

**Review by mattsherman — DISMISSED** [review]
Many of the release notes entries feel like "Dev" ones, so I've suggested marking them as such.

This brings up the larger question for me though... who are the release notes geared towards? If site admins are using these to help better understand what has changed, we should make sure that all non-Dev entries clearly state what has changed from an end user/admin perspective. And, maybe we should consider grouping all the Dev entries separately at the bottom of the list?

**Comment by timmyc** [comment]
Thanks @psealock and @mattsherman 

> Should we take the opportunity to move to an RC-* release process?

Yeah I kind of just fell into the old version release mode. I talked this over with @ObliviousHarmony today and since the release is cut, we are just going to stay with it this time around, but for 4.3 I agree we should do `1.3.0-beta1` just like core does, and then do one final 1.3.0 release for the final.

> This brings up the larger question for me though... who are the release notes geared towards?

I think the first and probably most important consumer of the changelog would be the core release flow ( Proton ). Beyond that, I imagine some developers read them, but store owners/merchants probably not.

I did re-organize the changelog to de-prioritize the dev items ( I had done them chronologically while assembling it this time ).

**Review by psealock — APPROVED** [review]
Thanks for the changes @timmyc :shipit:

**Pull request merged**